noxfile.py

```diff
@@ -0,0 +1,166 @@
+
+
+@nox.session(name="unit-tests-python", python=parse_supported_python_versions())
+def unit_tests(session: Session) -> None:
+    """Run unit tests and generate coverage report."""
+    install_requirements(session)
+    cmd = "pytest --log-level=DEBUG --cov=cachi2 --cov-config=pyproject.toml --cov-report=term --cov-report=html --no-cov-on-fail tests/unit"
+    session.run(*cmd.split(), *session.posargs)
+
+
+@nox.session(name="coverage-html")
+def coverage_html(session: Session) -> None:
+    """Open the coverage report in the browser."""
+    session.install("coverage")
+    session.run("coverage", "html")
+    session.run("xdg-open", "htmlcov/index.html", external=True)
+
+
+def _run_integration_tests(session: Session, env: dict[str, str]) -> None:
+    install_requirements(session)
+    netrc = "machine 127.0.0.1 login cachi2-user password cachi2-pass"
+    default_env = {"CACHI2_TEST_NETRC_CONTENT": os.getenv("CACHI2_TEST_NETRC_CONTENT", netrc)}
+    default_env.update(env)
+    cmd = "pytest --log-cli-level=WARNING tests/integration"
+    session.run(*cmd.split(), *session.posargs, env=default_env)
+
+
+@nox.session(name="integration-tests")
+def integration_tests(session: Session) -> None:
+    """Run integration tests only for the affected code base in the current branch."""
+    _run_integration_tests(session, {})
+
+
+@nox.session(name="all-integration-tests")
+def all_integration_tests(session: Session) -> None:
+    """Run all integration tests that are available."""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_RUN_ALL_INTEGRATION_TESTS": "true",
+            "CACHI2_TEST_LOCAL_PYPISERVER": "true",
+            "CACHI2_TEST_LOCAL_DNF_SERVER": "true",
+        },
+    )
+
+
+@nox.session(name="generate-test-data")
+def generate_test_data(session: Session) -> None:
+    """Run all integration tests that are available and update SBOMs."""
+    _run_integration_tests(
+        session,
+        {
+            "CACHI2_GENERATE_TEST_DATA": "true",
+            "PYTEST_ADDOPTS": "-k test_e2e",
+        },
+    )
+
+
+@nox.session(name="pip-compile")
+def pip_compile(session: Session) -> None:
+    """Update requirements.txt and requirements-extras.txt files."""
+    PWD = session.env["PWD"]
+    # git must be installed in the image due to setuptools-scm that has it as a direct dependency
+    pip_compile_cmd = """
+    apk add git && \
+    pip3 install pip-tools && \
+    pip-compile \
+        --generate-hashes \
+        --output-file=requirements.txt \
+        --rebuild \
+        pyproject.toml && \
+    pip-compile \
+        --all-extras \
+        --generate-hashes \
+        --output-file=requirements-extras.txt \
+        --rebuild \
+        pyproject.toml
+    """
+    cmd = f"podman run --rm --volume {PWD}:/cachi2:rw,Z --workdir /cachi2 docker.io/library/python:3.9-alpine sh -c {pip_compile_cmd}"
+    session.run(*cmd.split(), pip_compile_cmd, silent=True, external=True)
```

Comment:
If not for anything else, running podman from nox using a session runner is exotic enough to justify not silencing the output. The other reason being that you actually want to see what pip-compile is doing.